src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -5017,10 +5022,25 @@ namespace ZoneTempPredictorCorrector {
 				ZoneMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).InletNode( NodeNum ) ).MassFlowRate / ZoneMult;
 			} // NodeNum
 
+			ExhaustNodeFlag = false;
 			for ( NodeNum = 1; NodeNum <= ZoneEquipConfig( ZoneEquipConfigNum ).NumExhaustNodes; ++NodeNum ) {
-				ExhMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ).MassFlowRate / ZoneMult;
+				ExhaustListFlag = false;
+				for ( iList = 1; iList <= ZoneEquipList( ZoneNum ).NumOfEquipTypes; ++iList ) {
+					if ( ( ZoneEquipList( ZoneNum ).EquipType_Num( iList ) == ZoneExhaustFan_Num ) && 
+						( ZoneEquipList( ZoneNum ).EquipData( iList ).NumInlets > 0 && ZoneEquipList( ZoneNum ).EquipData( iList ).InletNodeNums( 1 ) == ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ) ) {
+						if ( !( ZoneEquipConfig( ZoneEquipConfigNum ).AirLoopNum > 0 || ZoneAirMassFlow.EnforceZoneMassBalance || AirflowNetworkNumOfExhFan > 0 ) ) {
+							ExhaustNodeFlag = true;
+							ExhaustListFlag = true;
+						}
+					}
+				}
+				if ( !ExhaustListFlag ) {
+					ExhMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ).MassFlowRate / ZoneMult;
+				}
 			} // NodeNum
-			ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExhBalanced; // Balanced exhaust flow assumes there are other flows providing makeup air such as mixing or infiltration, so subtract it here
+			if ( !ExhaustNodeFlag ) {
+				ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExhBalanced; // Balanced exhaust flow assumes there are other flows providing makeup air such as mixing or infiltration, so subtract it here
+			}
 
 			if ( ZoneEquipConfig( ZoneEquipConfigNum ).ReturnAirNode > 0 ) {
 				TotExitMassFlowRate = ExhMassFlowRate + Node( ZoneEquipConfig( ZoneEquipConfigNum ).ReturnAirNode ).MassFlowRate / ZoneMult;
```

Comment:
@miwitte I think the fix is correct. You can see testing results to show correct outputs.

You are right, CorrectZoneContaminants should be modified in the same way. I think the fix should be verified first. If accepted, I am going to do the same way in CorrectZoneContaminants.

I think you also notice CorrectZoneAirTemp uses different approach to handle zone balance components and does not have this item. The component process is accomplished in the CalcZoneSums. I can not tell why from my memory. I notice the difference before this issue. I plan to create a new function to handle moisture component loads, similar to CalcZoneSums. It will take time and needs detailed consideration. I hope this will be a new feature for FY18 to ensure all component loads are mapped correctly in all zone variables, such as temperature, moisture, CO2, and contaminant.